delete_purchases/delete_purchases.py

```diff
@@ -39,51 +28,83 @@ def get_purchase_lines(api):
     request_body = {"query": queries.GET_PURCHASE_LINES}
     return api.request(request_body)["data"]
 
+def get_purchase_line_etag(id,api):
+    request_body = {"query": queries.GET_PURCHASE_LINE_ETAG, "variables": {"id": id}}
+    purchase_line = api.request(request_body)["data"]
+    _etag = purchase_line["purchaseOrderLine"]["_etag"]
+    return _etag
 
 def get_purchases(api):
     request_body = {"query": queries.GET_PURCHASES}
     return api.request(request_body)["data"]
 
+def build_list_aboms_items(purchase_order_lines):
+    list_of_pos = set()
+    for purchase_order_line in purchase_order_lines["purchaseOrderLines"]["edges"]:
+        if "partInventories" in purchase_order_line["node"]:
+            if any(part_inventory["installed"] for part_inventory in purchase_order_line["node"]["partInventories"]) or any(part_inventory["kitted"] for part_inventory in purchase_order_line["node"]["partInventories"])or any(part_inventory["received"] for part_inventory in purchase_order_line["node"]["partInventories"]):
+                po_id = purchase_order_line["node"]["purchaseOrder"]["id"]
+                list_of_pos.add(po_id)
+    return list(list_of_pos)
+
+def build_list_receipts(receipts):
+    LIST_OF_POS = []
+    for receipt in receipts["receipts"]["edges"]:
+        polines = receipt["node"]["purchaseOrderLines"]
+        for poline in polines:
+            poline_id = poline["id"]
+            LIST_OF_POS.append(poline_id)
+    return LIST_OF_POS
 
 def delete_receipts(receipts, api):
     for receipt in receipts["receipts"]["edges"]:
         receipt_id = receipt["node"]["id"]
-        logger.info(f'Deleting receipt id: {receipt_id}')
+        print('Deleting receipt id: ',receipt_id)
         etag = receipt["node"]["_etag"]
         request_body = {
             "query": queries.DELETE_RECEIPT,
             "variables": {"id": receipt_id, "etag": etag},
         }
         api.request(request_body)
 
-
 def delete_purchase_lines(purchase_lines, api):
+    LIST_POS = set()
```

Comment:
I'd recommend reusing `PURCHASES_TO_SKIP`? Also capitalized text generally means a global variable, which `LIST_POS` is just used within this function.